### ISSUE TITLE
Android support for link to announce as link and not as button in A11Y

### ISSUE THREAD

**Issue by MaruQuarky** [flutter/flutter#102535]
## Use case

On the current documentation of Flutter for [link property](https://api.flutter.dev/flutter/semantics/SemanticsProperties/link.html) of [SemanticsProperties class](https://api.flutter.dev/flutter/semantics/SemanticsProperties-class.html), the links were read as button not as links.

> If non-null, indicates that this subtree represents a link.
> 
> iOS's VoiceOver provides users with a unique hint when a link is focused. **Android's Talkback will announce a link hint the same way it does a button**.

As documented in the [Android Accessibility Help page](https://support.google.com/accessibility/android/answer/6378148?hl=en), the design now allows links and hyperlinks to be read as links:

> ### Design
> 
> When using [TalkBack](https://support.google.com/accessibility/android/answer/6283677) to speak text containing a hyperlink, a well implemented user interface produces an auditory icon as the hyperlinked text is spoken, and the linked text appears in the "Links" sub-menu of TalkBack's [local context menu](https://support.google.com/accessibility/android/answer/6007066).

Previously, a [GitHub ticket](https://github.com/flutter/flutter/issues/60039) was filed regarding this issue but was close because it was WAI per [this comment](https://github.com/flutter/flutter/issues/60039#issuecomment-648478141):

> This was addressed in another issue also for `customer: money (g3)` : https://github.com/flutter/flutter/issues/36306
> Link was specifically used as a button since android doesn't offer a link option in the a11y class name.
> Further reference:
> 
> * https://github.com/flutter/engine/pull/12453/files
> * https://developer.android.com/reference/android/widget/Button
> 
> I think this is WAI since we intentionally did this because Android doesn't have an explicit 'link'.

## Proposal

Support to announce links and hyperlinks as links not as buttons in A11Y.

**Comment by darshankawar** [comment]
Thanks for the report. Treating this as a proposal and labeling for further insights from the team.

**Comment by chunhtai** [comment]
It looks like android can only use URLSpan to make a label of node be treated as a link, but it requires a url. So in order for it to work, flutter need to also provide a url string instead of just isLink:true.

while iOS has UIAccessibilityTraitLink, it also has NSLinkAttributeName. It it has similar effect to voiceover. has We can also think about adding a new StringAttribute for url link

**Comment by chunhtai** [comment]
It looks like the NSLinkAttributeName is not for a11y purposes. We may need to come up a new API semantics node to store the link url to support android link

**Comment by chunhtai** [comment]
DD flutter.dev/go/a11y-links

**Comment by gmackall** [comment]
@chunhtai given that you have a design doc for this currently, is it fine to re-assign you?

**Comment by reidbaker** [comment]
@chunhtai let me know if assigning to you is the wrong call.

**Comment by Piinks** [comment]
@hangyujin @chunhtai was this fixed? I thought it had been...

**Comment by chunhtai** [comment]
no, I believe we only fixes for web, we still need to use the newly added semantics url to wrap the contentdescription with urlSpan in android a11y bridge

**Comment by reidbaker** [comment]
@chunhtai is this part of the work you/your team have planned for q4?

**Comment by reidbaker** [comment]
I will talk to @mariamhas but my guess is no. Are you saying this bug needs to be reassigned to team-android from team-accessibility?

**Comment by chunhtai** [comment]
The change will need to be made in AccessibilityBridge.java, basically wrap the contentdescription with the urlspan, but I am not sure if I have time to get to it this quarter.

> are you saying this bug needs to be reassigned to team-android from team-accessibility

I think leaving it team-accessibility is fine, though do you know whether your team owns accessibility code in Android embedding?

**Comment by reidbaker** [comment]
I think the accessibility team owns that code. Best I can find is that the ownership of that code was unknown when this document was authored. [go/flutter-android-charter](http://goto.google.com/flutter-android-charter)
